### PR TITLE
[#ICC-238] add get user session endpoint

### PR DESCRIPTION
**Pull request by fabriziopapi**
Add a new enpoint to retrieve the user session activation status. This endpoint will proxies the access to redis cache.
The io-fn-message will use it to select the select the right notification template to puth to the user.

#### List of Changes

add new endpoint to session controller

#### Motivation and Context

The push notification template used by reminder must be different for logged and non-logged user. This endpoint will avoid to perform the specific logic and redis cache access in the azure functions.

#### How Has This Been Tested?

unit test

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/controllers/__tests__/sessionLockController.test.ts [hunk 1]
```diff
@@ -45,7 +103,7 @@ describe("SessionLockController#lockUserSession", () => {
       mockRedisUserMetadataStorage
     );
 
-    const response = await controller.lockUserSession(req);
+    const response = await controller.getUserSession(req);
```
Comment (gquadrati) [inline src/controllers/__tests__/sessionLockController.test.ts:106]: Why this change?

File: src/app.ts [hunk 2]
```diff
@@ -1110,6 +1110,15 @@ function registerSessionAPIRoutes(
       userMetadataStorage
     );
 
+    app.get(
```
Comment (AleDore) [inline src/app.ts:1113]: this route is actually mounted when `FF_ENABLE_SESSION_LOCK_ENDPOINT` is enabled. Are we sure this is the correct behaviour?

Comment (gquadrati) [reply src/app.ts:1113]: You're right, this variable is based on `IS_APPBACKENDLI` but it's more correct to create a new one 
https://github.com/pagopa/io-backend/blob/88b17f878f35f6405f74a157b91fe6cb974d5b82/src/config.ts#L868

Comment (gquadrati) [reply src/app.ts:1113]: Renamed `FF_ENABLE_SESSION_LOCK_ENDPOINT` to a more generic `FF_ENABLE_SESSION_ENDPOINTS`

File: src/controllers/__tests__/sessionLockController.test.ts (hunk 1 above)
Comment (gquadrati) [reply src/controllers/__tests__/sessionLockController.test.ts]: reverted in [765d30d](https://github.com/pagopa/io-backend/pull/929/commits/765d30d4bac578e073586619f853242ce7b5d0c4)

File: api_session.yaml [hunk 3]
```diff
@@ -85,6 +114,13 @@ definitions:
     properties:
       message:
         type: string
+  UserSession:
```
Comment (BurnedMarshal) [inline api_session.yaml:117]: What if we rename this object with
 ```suggestion
   UserSessionStatus:
```
?

File: src/controllers/__tests__/sessionLockController.test.ts [hunk 4]
```diff
@@ -24,17 +24,75 @@ const mockSetBlockedUser = jest
...
+
+    expect(res.status).toHaveBeenCalledWith(400);
+  });
+
+  it("should fail on get error", async () => {
+    const req = mockReq({ params: { fiscal_code: aFiscalCode } });
+    const res = mockRes();
+
+    mockUserHasActiveSessions.mockImplementationOnce(async () =>
+      E.left("any error")
```
Comment (BurnedMarshal) [inline src/controllers/__tests__/sessionLockController.test.ts:63]: There is a type error in here. `userHasActiveSessions` returns a `Promise<Either<Error, boolean>>`. This should be like 
```suggestion
      E.left(new Error("any error"))
```

File: src/controllers/__tests__/sessionLockController.test.ts [hunk 5]
```diff
@@ -24,17 +24,75 @@ const mockSetBlockedUser = jest
...
+      mockRedisUserMetadataStorage
+    );
+
+    const response = await controller.getUserSession(req);
+    response.apply(res);
+
+    expect(res.status).toHaveBeenCalledWith(400);
+  });
+
+  it("should fail on get error", async () => {
```
Comment (BurnedMarshal) [inline src/controllers/__tests__/sessionLockController.test.ts:58]: Is more than a single get, maybe a more specific description remove ambiguity for example `should fail if userHasActiveSessions  returns an error`

File: api_session.yaml [hunk 6]
```diff
@@ -8,6 +8,35 @@ basePath: /api/v1
...
+          examples:
+            application/json:
+              "active": "true"
+        "400":
+          description: Bad request
+        "401":
+          description: Token null or invalid.
+        "404":
+          description: User not found.
+        "429":
```
Comment (BurnedMarshal) [inline api_session.yaml:34]: Where is configured the Throttling?

File: api_session.yaml [hunk 7]
```diff
@@ -8,6 +8,35 @@ basePath: /api/v1
...
+        - $ref: "#/parameters/FiscalCode"
+        - $ref: "#/parameters/Token"
+      responses:
+        "200":
+          description: Success.
+          schema:
+            $ref: "#/definitions/UserSession"
+          examples:
+            application/json:
+              "active": "true"
```
Comment (BurnedMarshal) [inline api_session.yaml:27]:
```suggestion
              "active": true
```

**Review by BurnedMarshal — CHANGES_REQUESTED** [review]

File: api_session.yaml (hunk 3 above)
Comment (fabriziopapi) [reply api_session.yaml:117]: I named it `UserSession` becouse in the future we could expand it with other fields

Comment (BurnedMarshal) [reply api_session.yaml:117]: Nice, maybe could be better `UserSessionInfo`

File: src/controllers/__tests__/sessionLockController.test.ts (hunk 5 above)
Comment (fabriziopapi) [reply src/controllers/__tests__/sessionLockController.test.ts:58]: you are right. thank you!

File: api_session.yaml (hunk 6 above)
Comment (fabriziopapi) [reply api_session.yaml:34]: nowhere (for now). I added it for future implementation.

File: api_session.yaml (hunk 3 above)
Comment (fabriziopapi) [reply api_session.yaml:117]: you are right. I'll fix it. Thank you!

**Review by gquadrati — APPROVED** [review]
LGTM

**Review by AleDore — APPROVED** [review]
LGTM
